src/Microsoft.Health.Dicom.Core/Features/Query/QueryService.cs

```diff
@@ -44,7 +53,11 @@ public class QueryService : IQueryService
 
             ValidateRequestIdentifiers(message);
 
-            QueryExpression queryExpression = _queryParser.Parse(message);
+            IReadOnlyList<CustomTagStoreEntry> customTags = await _customTagStore.GetCustomTagsAsync(null, cancellationToken);
+
+            HashSet<CustomTagFilterDetails> supportedCustomTags = RetrieveSuupportedCustomTagsForQueryResourceType(customTags, message.QueryResourceType);
```

Comment:
Also instead of getting this and passing it to QueryParser, see if you can get the QueriableTags

Reply:
I think the same think we talked with Peng. We already have CustomTagService. Can we create IndexedTagService and make its problem to combine and get the full list and use that service everywhere?